backend/internal/server/apis/schedule/games/games.go

```diff
@@ -0,0 +1,286 @@
+	log.Info("Reading body\n")
+	seasonName, numberOfGamesPerTeam, err := readBody(c)
+	if err != nil {
+		log.Error("Error reading body: %v\n", err)
+		return err
+	}
+
+	if seasonName == "" {
+		log.Error("Season file name is empty\n")
+		return responder.BadRequest(c, "Season file name is empty")
+	}
+
+	if numberOfGamesPerTeam == 0 {
+		log.Error("Number of games per team is 0\n")
+		return responder.BadRequest(c, "Number of games per team is 0")
+	}
+
+	log.Info("Reading config file for season\n")
+	seasonConfig, err := parser.SeasonConfig(seasonName)
+	if err != nil {
+		log.Error("Error reading file: %v\n", err)
+		return responder.BadRequest(c, "Error reading file")
+	}
+
+	log.Info("Generating games\n")
+	season, err := generateGames(seasonConfig.Leagues, numberOfGamesPerTeam)
+
+	log.Info("Assigning ice times\n")
+	games, err := assignTimes(seasonConfig.IceTimes, season, numberOfGamesPerTeam)
+	if err != nil {
+		log.Error("Error assigning ice times: %v\n", err)
+		return responder.BadRequest(c, "Error assigning ice times")
+	}
+
+	log.Info("Optimizing schedule\n")
+	optimizeSchedule(games)
+
+	log.Info("Writing csv\n")
+	err = csv.GenerateCsv(games, "schedule.csv")
+	if err != nil {
+		log.Error("Error writing csv: %v\n", err)
+		return responder.InternalServerError(c)
+	}
+
+	//log.Info("Saving to database\n")
+	//err = save.ToDb(c, games)
+	//if err != nil {
+	//	log.Error("Error saving to database: %v\n", err)
+	//	return responder.InternalServerError(c)
+	//}
+
+	return responder.Ok(c, "Schedule generated at schedule.csv and saved to database")
+}
+
+func readBody(c *fiber.Ctx) (string, int, error) {
+	type BodyDto struct {
+		SeasonName           string `json:"seasonName"`
+		NumberOfGamesPerTeam int    `json:"numberOfGamesPerTeam"`
+	}
+	body := c.Body()
+	var bodyDto BodyDto
+	err := json.Unmarshal(body, &bodyDto)
+	if err != nil {
+		return "", 0, responder.BadRequest(c, "Error reading body")
+	}
+
+	return bodyDto.SeasonName, bodyDto.NumberOfGamesPerTeam, nil
+}
+
+func optimizeSchedule(games []models.Game) {
+	if len(games) == 0 {
+		log.Info("No games to optimize")
+		return
+	}
+	log.Info("Pre-optimization analysis")
+	seasonStats, teamStats := analysis.RunTimeAnalysis(games)
+
+	// Need to make sure games are balanced in
+	// - Early / late
+	// - Days between games
+	balanceCount := getBalanceCount(&teamStats)
+	lastBalanceCount := -1
+
+	for count := 0; balanceCount != lastBalanceCount && count < 25; count++ {
+		optimize.Schedule(games, seasonStats, teamStats)
+
+		log.Info("Post-optimization analysis")
+		seasonStats, teamStats = analysis.RunTimeAnalysis(games)
+
+		lastBalanceCount = balanceCount
+		balanceCount := getBalanceCount(&teamStats)
+
+		log.Info("Balanced count: %v\n", balanceCount)
+	}
+}
+
+func generateGames(leagues []models.League, numberOfGamesPerTeam int) (models.Season, error) {
+	if len(leagues) == 0 {
+		return models.Season{}, errors.New("no leagues to games games for")
```

Comment:
Kind of nitpicky, but this error isn't coherent

Reply:
Fixed:)